biocrnpyler/chemical_reaction_network.py

```diff
@@ -137,16 +164,32 @@ def pretty_print(self, show_rates = True, show_material = True, show_attributes
         show_rates toggles whether reaction rate functions are printed
         """
 
-        txt = f"Species ({len(self.species)}) = "+"{"
-        for sind in range(len(self.species)):
-            s = self.species[sind]
-            txt += f"{sind}. "+s.pretty_print(show_material = show_material, show_attributes = show_attributes, show_initial_condition = show_initial_condition, **kwargs) + ", "
-        txt = txt[:-2] + '}\n'
+        txt = "Species"+ f"(N = {len(self.species)}) = "+"{\n"
+
+        ics = lambda s: self.initial_condition_dict[s] if s in self.initial_condition_dict else 0
+
+        species_sort_list = [(parameter_to_value(ics(s)), s) for s in self.species]
+        species_sort_list.sort()
+        species_sort_list.reverse()
+        for sind, (init_cond, s) in enumerate(species_sort_list):
+            init_cond = ics(s) 
+
+            txt += s.pretty_print(show_material = show_material, show_attributes = show_attributes, **kwargs)
+
+            if show_initial_condition:
+                txt += f" (@ {parameter_to_value(init_cond)}),  "
```

Comment:
do you have a helper function for such pretty_print in the ParameterDatabase class already?
If not, it might be better to write a slightly more general code in ParameterDatabase and call it with initial condition parameterization. Then such code can be reused elsewhere

Reply:
The reason I use this parameter_to_value helper function is because the initial conditions can be parameters, or numbers, and this takes care of both cases.

Reply:
I see, in that case, my comment was not clear. Can you hide `more lines of code` inside that helper function?

Reply:
Don't think so. Parameters don't know what they are used for, and in some cases the init_cond is actually just a number. I don't want to return a string, because I use that helper function elsewhere when dealing with parameters which I want to treat as numbers.